Node-sass error

npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm ERR! code 1
npm ERR! path C:\Users\10016\WebstormProjects\new-design-1\node_modules\node-sass
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node scripts/build.js
npm ERR! Building: C:\Program Files\nodejs\node.exe C:\Users\10016\WebstormProjects\new-design-1\node_modules\node-gyp\bin\node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp verb cli [
npm ERR! gyp verb cli   'C:\\Program Files\\nodejs\\node.exe',
npm ERR! gyp verb cli   'C:\\Users\\10016\\WebstormProjects\\new-design-1\\node_modules\\node-gyp\\bin\\node-gyp.js',
npm ERR! gyp verb cli   'rebuild',
npm ERR! gyp verb cli   '--verbose',
npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! gyp verb cli   '--libsass_library='
npm ERR! gyp verb cli ]
npm ERR! gyp info using node-gyp@3.8.0
npm ERR! gyp info using node@17.7.0 | win32 | x64
npm ERR! gyp verb command rebuild []
npm ERR! gyp verb command clean []
npm ERR! gyp verb clean removing "build" directory
npm ERR! gyp verb command configure []
npm ERR! gyp verb check python checking for Python executable "python2" in the PATH
npm ERR! gyp verb `which` failed Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:13:12)
npm ERR! gyp verb `which` failed     at F (C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:68:19)
npm ERR! gyp verb `which` failed     at E (C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:80:29)
npm ERR! gyp verb `which` failed     at C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:89:16
npm ERR! gyp verb `which` failed     at C:\Users\10016\WebstormProjects\new-design-1\node_modules\isexe\index.js:42:5
npm ERR! gyp verb `which` failed     at C:\Users\10016\WebstormProjects\new-design-1\node_modules\isexe\windows.js:36:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:198:21)
npm ERR! gyp verb `which` failed  python2 Error: not found: python2
npm ERR! gyp verb `which` failed     at getNotFoundError (C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:13:12)
npm ERR! gyp verb `which` failed     at F (C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:68:19)
npm ERR! gyp verb `which` failed     at E (C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:80:29)
npm ERR! gyp verb `which` failed     at C:\Users\10016\WebstormProjects\new-design-1\node_modules\which\which.js:89:16
npm ERR! gyp verb `which` failed     at C:\Users\10016\WebstormProjects\new-design-1\node_modules\isexe\index.js:42:5
npm ERR! gyp verb `which` failed     at C:\Users\10016\WebstormProjects\new-design-1\node_modules\isexe\windows.js:36:5
npm ERR! gyp verb `which` failed     at FSReqCallback.oncomplete (node:fs:198:21) {
npm ERR! gyp verb `which` failed   code: 'ENOENT'
npm ERR! gyp verb `which` failed }
npm ERR! gyp verb check python checking for Python executable "python" in the PATH
npm ERR! gyp verb `which` succeeded python C:\Python310\python.EXE
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: Command failed: C:\Python310\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack   File "<string>", line 1
npm ERR! gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
npm ERR! gyp ERR! stack                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
npm ERR! gyp ERR! stack SyntaxError: Missing parentheses in call to 'print'. Did you mean print(...)?
npm ERR! gyp ERR! stack
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (node:child_process:398:12)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
npm ERR! gyp ERR! stack     at maybeClose (node:internal/child_process:1090:16)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! gyp ERR! System Windows_NT 10.0.19044
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\10016\\WebstormProjects\\new-design-1\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd C:\Users\10016\WebstormProjects\new-design-1\node_modules\node-sass
npm ERR! gyp ERR! node -v v17.7.0
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok
npm ERR! Build failed with error code: 1

--- a/app/templates/_package.json
+++ b/app/templates/_package.json
@@ -36,7 +36,7 @@
     "gulp-plumber": "^1.2.1",
     "gulp-postcss": "^8.0.0",
     <%_ if (includeSass) { -%>
-    "gulp-sass": "^4.0.2",
+    "gulp-sass": "^5.1.0",
     <%_ } -%>
     "gulp-size": "^3.0.0",
     "gulp-uglify": "^3.0.2",
@@ -45,6 +45,9 @@
     "mkdirp": "^0.5.1",
     <%_ } -%>
     "mocha": "^6.1.4",
+    <%_ if (includeSass) { -%>
+    "sass": "^1.58.3",
+    <%_ } -%>
     "yargs": "13.2.4"
   },
   "scripts": {

--- a/app/templates/gulpfile.js
+++ b/app/templates/gulpfile.js
@@ -9,6 +9,9 @@ const Modernizr = require('modernizr');
 const browserSync = require('browser-sync');
 const del = require('del');
 const autoprefixer = require('autoprefixer');
+<%_ if (includeSass) { -%>
+const sass = require('gulp-sass')(require('sass'));
+<%_ } -%>
 const cssnano = require('cssnano');
 const { argv } = require('yargs');
 
@@ -27,11 +30,11 @@ function styles() {
     sourcemaps: !isProd,
   })
     .pipe($.plumber())
-    .pipe($.sass.sync({
+    .pipe(sass.sync({
       outputStyle: 'expanded',
       precision: 10,
       includePaths: ['.']
-    }).on('error', $.sass.logError))
+    }).on('error', sass.logError))
   <%_ } else { -%>
   return src('app/styles/*.css', {
     sourcemaps: !isProd,

